src/Guardrails-Common/GR-Common.psm1

```diff
@@ -396,30 +396,61 @@ function Check-GAAuthenticationMethods {
                 $data = $response.Content
                 if ($null -ne $data -and $null -ne $data.value) {
                     $authenticationmethods = $data.value
-
-                    # To check if MFA is setup for a user, we're looking for either :
-                    #   #microsoft.graph.microsoftAuthenticatorAuthenticationMethod or
-                    #   #microsoft.graph.phoneAuthenticationMethod            
-                    foreach ($authmeth in $authenticationmethods) {
-                        if (($($authmeth.'@odata.type') -eq "#microsoft.graph.phoneAuthenticationMethod") -or `
-                                ($($authmeth.'@odata.type') -eq "#microsoft.graph.microsoftAuthenticatorAuthenticationMethod")) {
-                                # need to keep track of each GA mfa in counter and compare it to count
-                                $mfaCounter += 1
-                                # found atleast one auth method so we move to the next UPN 
-                                break 
-                            }
+                    
+                    # To check if MFA is setup for a user, we're looking for 4 types of authentication methods :
+                        # 1.  #microsoft.graph.microsoftAuthenticatorAuthenticationMethod
+                        # 2.  #microsoft.graph.phoneAuthenticationMethod    
+                        # 3.  #microsoft.graph.passwordAuthenticationMethod 
+                        # 4.  #microsoft.graph.emailAuthenticationMethod  
+                        
+                    # Condition: when there is only 1 authentication method found, usually MFA is not setup - it needs to be checked separately
+                    if ($authenticationmethods.Count -eq 1){
+                        if(($($authmeth.'@odata.type') -eq "#microsoft.graph.passwordAuthenticationMethod") ) {
+                            
+                            # need to keep track of each GA mfa in counter and compare it to count
+                            $mfaCounter += 1
+                                    
+                        }
                         else {
+                            # This message will be used for debugging
+                            Write-Host "Found only 1 authentication method with MFA not enabled"
                             # create an instance of inner list object
                             $GAUPNtemplate = [PSCustomObject]@{
                                 UPN  = $globalAdminAccount
                                 MFAStatus   = $false
                                 MFAComments = $hiddenUPN
                                 
                             }
-                            #add the list to GA MFA list
+                            # add the list to GA MFA list
                             $GAUPNsMFA += $GAUPNtemplate
                         }
                     }
+                    # Condition: when user's authentication methods > 1
```

Comment:
@dutt0  

We can make this code a little bit cleaner 

```
$targetAuthMethodTypes = @(
    "#microsoft.graph.phoneAuthenticationMethod",
    "#microsoft.graph.microsoftAuthenticatorAuthenticationMethod",
    "#microsoft.graph.passwordAuthenticationMethod",
    "#microsoft.graph.emailAuthenticationMethod"
)

$GAUPNtemplate = [PSCustomObject]@{
    UPN        = $globalAdminAccount
    MFAStatus  = $false
    MFAComments = $hiddenUPN
}

foreach ($authmeth in $authenticationmethods) {
    if ($targetAuthMethodTypes -contains $($authmeth.'@odata.type')) {
        $mfaCounter += 1
        break
    } else {
        # Clone the template and add to the list
        $GAUPNsMFA += $GAUPNtemplate.psobject.Copy()
    }
}
```



I think this introduces a bug what happens if first authMeth that comes in the iteration is none of the 4 auth methods, wouldn't this create a  GAUPNTemplate list and then in the next iteration it finds the one of the 4 auth methods and it also increases mfacounter 

So we end up posting message of mfa not enabled as well as showing message that it is enabled for that UPN